### DISCUSSION TITLE
Best tool to load web parts and content from Sharepoint.

### DISCUSSION THREAD

**Discussion by YuriyMorozyuk95** [microsoft/hack-together-teams#34] [Q&A] [answered]
Could you please recommend best tool to load web parts/contents from manual writenn on sharepoint, basically load headers, and contenent inside <p> tabs, inside web parts, etc.

If possible to make some preview of content will be nice.

Before I use CSOM for load documents and update metadata, I heard that a lot of new tools came, like PnP Framework, Microsoft Graph, etc to manage Sharepoint content from client.

I am using C# leanguage for this task. 
Thank you.

**Comment by waldekmastykarz** [comment, marked as answer]
If you're building a JS/TS-based app then PnPjs is the way to go: https://pnp.github.io/pnpjs/sp/clientside-pages/#load-pages. For C#-based solutions, you'd use the PnP Core library: https://pnp.github.io/pnpcore/using-the-sdk/pages-intro.html.
